#!/usr/bin/env python
PACKAGE = "qm_wbc"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp_arm_linear", double_t, 0, "Kp of arm linear", 150, 0.0, 20000)
gen.add("kd_arm_linear", double_t, 0, "Kd of arm linear", 40, 0.0, 1000)
gen.add("kp_arm_angular", double_t, 0, "Kp of arm angular", 150, 0.0, 1000)
gen.add("kd_arm_angular", double_t, 0, "Kd of arm angular", 40, 0.0, 1000)

gen.add("kp_arm_joint", double_t, 0, "Kp of arm joint", 150, 0.0, 1000)
gen.add("kd_arm_joint", double_t, 0, "Kd of arm joint", 40, 0.0, 1000)

gen.add("baseHeightKp", double_t, 0, "Kp of base height", 400, 0.0, 20000)
gen.add("baseHeightKd", double_t, 0, "Kd of base height", 140, 0.0, 20000)

gen.add("kp_base_linear", double_t, 0, "Kp of base linear", 400, 0.0, 20000)
gen.add("kd_base_linear", double_t, 0, "Kd of base linear", 140, 0.0, 20000)

gen.add("kp_base_angular", double_t, 0, "Kp of base angular", 400, 0.0, 500)
gen.add("kd_base_angular", double_t, 0, "Kd of base angular", 140, 0.0, 500)

gen.add("kp_arm_joint_1", double_t, 0, "Kp of arm linear", 4000, 0.0, 20000)
gen.add("kp_arm_joint_2", double_t, 0, "Kd of arm linear", 4200, 0.0, 20000)
gen.add("kp_arm_joint_3", double_t, 0, "Kp of arm angular", 4000, 0.0, 20000)
gen.add("kp_arm_joint_4", double_t, 0, "Kd of arm angular", 4000, 0.0, 20000)
gen.add("kp_arm_joint_5", double_t, 0, "Kp of arm angular", 4200, 0.0, 20000)
gen.add("kp_arm_joint_6", double_t, 0, "Kd of arm angular", 6000.0, 0.0, 20000)
gen.add("kp_arm_joint_7", double_t, 0, "Kd of arm angular", 6000.0, 0.0, 20000)


gen.add("kd_arm_joint_1", double_t, 0, "Kp of arm linear", 75, 0.0, 20000)
gen.add("kd_arm_joint_2", double_t, 0, "Kd of arm linear", 75, 0.0, 20000)
gen.add("kd_arm_joint_3", double_t, 0, "Kp of arm angular", 75, 0.0, 20000)
gen.add("kd_arm_joint_4", double_t, 0, "Kd of arm angular", 75, 0.0, 20000)
gen.add("kd_arm_joint_5", double_t, 0, "Kp of arm angular", 75, 0.0, 20000)
gen.add("kd_arm_joint_6", double_t, 0, "Kd of arm angular", 75, 0.0, 20000)
gen.add("kd_arm_joint_7", double_t, 0, "Kd of arm angular", 75, 0.0, 20000)


gen.add("d_arm_1", double_t, 0, "desired angle of arm joint 1", 0.0 , -6.28, 6.28)
gen.add("d_arm_2", double_t, 0, "desired angle of arm joint 2", 0.26, -6.28, 6.28)
gen.add("d_arm_3", double_t, 0, "desired angle of arm joint 3", 3.14, -6.28, 6.28)
gen.add("d_arm_4", double_t, 0, "desired angle of arm joint 4", -2.27, -6.28, 6.28)
gen.add("d_arm_5", double_t, 0, "desired angle of arm joint 5", 0.0 , -6.28, 6.28)
gen.add("d_arm_6", double_t, 0, "desired angle of arm joint 6", 0.96,  -6.28, 6.28)
gen.add("d_arm_7", double_t, 0, "desired angle of arm joint 7", 1.57,  -6.28, 6.28)


gen.add("kp_ee_linear_x", double_t, 0, "Kp of arm linear", 3000.0, 0.0, 20000) # 6000
gen.add("kp_ee_linear_y", double_t, 0, "Kd of arm linear", 3000.0, 0.0, 20000)
gen.add("kp_ee_linear_z", double_t, 0, "Kp of arm angular", 3000.0, 0.0, 20000)

gen.add("kd_ee_linear_x", double_t, 0, "Kp of arm linear", 75.0, 0.0, 20000) # 100
gen.add("kd_ee_linear_y", double_t, 0, "Kd of arm linear", 75.0, 0.0, 20000)
gen.add("kd_ee_linear_z", double_t, 0, "Kp of arm angular", 75.0, 0.0, 20000)

gen.add("d_ee_x", double_t, 0, "desired angle of arm joint 1", 0.57, -1, 1) # 0.52
gen.add("d_ee_y", double_t, 0, "desired angle of arm joint 2", 0.09, -1, 1)
gen.add("d_ee_z", double_t, 0, "desired angle of arm joint 3", 0.41, -1, 1) # 0.44

gen.add("kp_ee_angular_x", double_t, 0, "Kp of arm linear", 2000.0, 0.0, 20000)
gen.add("kp_ee_angular_y", double_t, 0, "Kd of arm linear", 2000.0, 0.0, 20000)
gen.add("kp_ee_angular_z", double_t, 0, "Kp of arm angular", 2000.0, 0.0, 20000)

gen.add("kd_ee_angular_x", double_t, 0, "Kp of arm linear", 75, 0.0, 20000)
gen.add("kd_ee_angular_y", double_t, 0, "Kd of arm linear", 75, 0.0, 20000)
gen.add("kd_ee_angular_z", double_t, 0, "Kp of arm angular", 75, 0.0, 20000)

gen.add("da_ee_z", double_t, 0, "desired angle of arm joint 3", -1.57, -3.14, 3.14) # -1.57
gen.add("da_ee_y", double_t, 0, "desired angle of arm joint 2", 0, -3.14, 3.14)
gen.add("da_ee_x", double_t, 0, "desired angle of arm joint 1", -1.57, -3.14, 3.14) # -1.57

gen.add("kt", double_t, 0, "desired angle of arm joint 1", 1, 0., 1)

exit(gen.generate(PACKAGE, "wbcweight", "WbcWeight"))